src/components/PipelineRun/PipelineRunListView/PipelineRunListRow.tsx

```diff
@@ -45,7 +55,46 @@ const BasePipelineRunListRow: React.FC<React.PropsWithChildren<BasePipelineRunLi
   if (!obj.metadata?.labels) {
     obj.metadata.labels = {};
   }
-  const applicationName = obj.metadata?.labels[PipelineRunLabel.APPLICATION];
+  const labels = obj.metadata.labels;
+  const applicationName = labels?.[PipelineRunLabel.APPLICATION];
+  const gitProvider = obj.metadata.annotations?.[PipelineRunLabel.COMMIT_PROVIDER_LABEL];
+  const repoOrg = labels?.[PipelineRunLabel.COMMIT_REPO_ORG_LABEL];
+  const repoURL = labels?.[PipelineRunLabel.COMMIT_REPO_URL_LABEL];
+  const prNumber = labels?.[PipelineRunLabel.PULL_REQUEST_NUMBER_LABEL];
+  const eventType = labels?.[PipelineRunLabel.COMMIT_EVENT_TYPE_LABEL];
+  const commidId = labels?.[PipelineRunLabel.COMMIT_LABEL];
+
+  const getTriggerredByColumnData = useCallback(() => {
+    let icon = null,
+      text = ``,
+      link = `https://${gitProvider}.com/${repoOrg}/${repoURL}`;
+    const commitDetails = {
+      text: commidId?.substring(0, 7),
+      link: `${link}/commit/${commidId}`,
+    };
+    if (eventType === PipelineRunEventType.PUSH || eventType === PipelineRunEventType.RETEST) {
+      icon = <CommitIcon isPR={false} className="sha-title-icon" />;
+    } else if (eventType === PipelineRunEventType.PULL) {
+      icon = <CommitIcon isPR={true} className="sha-title-icon" />;
+      text = `${repoOrg}/${repoURL}/${prNumber}`;
+      link = `${link}/pull/${prNumber}`;
+    }
+    return (
+      <>
+        {icon}
+        {eventType === PipelineRunEventType.PULL && (
+          <ExternalLink href={link} text={<Truncate content={text} />} hideIcon={true} />
+        )}
+        {eventType ? (
```

Comment:
checking event type twice to show two labels or just one label?
if one label, codes like this:
  {eventType === PipelineRunEventType.PULL ? (
    <ExternalLink href={link} text={<Truncate content={text} />} hideIcon={true} />
  ) : eventType ? (
    <Label color="blue">
      <ExternalLink href={link} text={text} />
    </Label>
  ) : (
    <span>-</span>
  )}
if two labels, ignore my comments